gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaStreamingExtractor.java

```diff
@@ -323,8 +323,10 @@ public KafkaStreamingExtractor(WorkUnitState state) {
   @Override
   public S getSchema() {
     try {
-      Schema schema = (Schema) this._schemaRegistry.get().getLatestSchemaByTopic(this.topicPartitions.get(0).getTopicName());
-      return (S) schema;
+      if(this._schemaRegistry.isPresent()) {
+        return (S)(Schema) this._schemaRegistry.get().getLatestSchemaByTopic(this.topicPartitions.get(0).getTopicName());
+      }
+      return (S) this.topicPartitions.iterator().next().getTopicName();
```

Comment:
return null? Don't understand why we are returning topic name.

Reply:
It's because many converter does not accept null in convertSchema() method. And simply return null will cause issue there. I just use the same behavior as KafkaSimpleStreamingExtractor where if the schema registry is null, we just return topic name as schema